### PR TITLE
Adds Git workflow scripts

### PR DESCRIPTION
**Pull request by mrjoro**
Adds some Git workflow scripts based on those created by @camelburrito.

Once this is in we can add some references to it in the getting started guides.

cc @erwinmombay

### CODE REVIEW

**Comment by mrjoro** [comment]
FYI I added a missing --rebase to the git pull in git sync.  I'd mentioned before that I wasn't sure if we needed the --rebase there, but I realized that since we're updating the master branch and we don't typically make changes on/push the local master branch that's fine.

I'm still not sure about the "git rebase master" at the end when you are in the middle of a PR since rebases can mess with existing commits.

File: contributing/git_helpers.sh [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+# then pushes the updates to your remote master branch and the local
+# branch you ran this command from.
+git_sync() {
+  local branch_name=$(git symbolic-ref -q HEAD)
+  branch_name=${branch_name##refs/heads/}
+  branch_name=${branch_name:-HEAD}
+  echo "Checking out your master branch"
+  git checkout master
+  echo "Updating your local master branch with the latest changes (pulling upstream/master)"
+  git pull --rebase upstream master
```
Comment (kmh287) [inline contributing/git_helpers.sh]: Any reason not to just do: `git pull upstream master` ? Is there a reason to rebase (here and on the line below) if users aren't committing directly to their repos' master branches?

File: contributing/git_helpers.sh [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+  git rebase -i
+  echo "Updating your remote repository with the latest changes (pushing origin/master)"
+  git push origin master -f
```
Comment (kmh287) [inline contributing/git_helpers.sh]: If the rebase is removed then there's no need to force the push.

File: contributing/git_helpers.sh (hunk 1 above)
Comment (mrjoro) [inline contributing/git_helpers.sh]: @camelburrito and @erwinmombay for this comment and below :)

Comment (camelburrito) [inline contributing/git_helpers.sh]: We cant tell what users do with their master - this is a little safer

Comment (mrjoro) [inline contributing/git_helpers.sh]: I'll go ahead and merge this in; if it causes problems we can change it.

**Pull request merged**